src/c++/tests/cc_client_test.cc

```diff
@@ -1300,6 +1300,42 @@ TYPED_TEST_P(ClientTest, LoadWithFileOverride)
         << "expect model " << override_name << " version " << vr.first
         << " readiness: " << vr.second;
   }
+
+  // Request to load the model with override config in original name
+  // Send the config with wrong format
+  config =
+      "\"parameters\": {\"config\": {{\"backend\":\"onnxruntime\", "
+      "\"version_policy\":{\"specific\":{\"versions\":[2]}}}}}";
+  err = this->LoadModel(model_name, config);
+  ASSERT_FALSE(err.IsOk()) << "Expect LoadModel() to fail";
+
+  // Send the config with the correct format
+  config =
+      "{\"backend\":\"onnxruntime\", "
+      "\"version_policy\":{\"specific\":{\"versions\":[2]}}}";
+  err = this->LoadModel(model_name, config);
+
+  // check that the model with original name is changed
+  expected_version_ready = {{"1", false}, {"2", true}, {"3", false}};
+  for (const auto& vr : expected_version_ready) {
+    bool ready = false;
+    err = this->client_->IsModelReady(&ready, model_name, vr.first);
+    ASSERT_TRUE(err.IsOk())
+        << "failed to get version readiness: " << err.Message();
+    ASSERT_EQ(ready, vr.second) << "expect model " << model_name << " version "
+                                << vr.first << " readiness: " << vr.second;
+  }
+
+  // Sanity check readiness of the different named model
```

Comment:
I don't understand what this is checking and why. Could you clarify in the comments?

Reply:
This just some check to see if the previous action only affects that specific model. On second thought, this should be covered from the testing above and maybe we can just remove this part?

Reply:
I'm less familiar with these tests, so not sure if we'd expect it to affect that model. Like is the naming is the same as the original/override? Totally up to you, you'd know this test better.